### DISCUSSION TITLE
How to establish Conditional required

### DISCUSSION THREAD

**Discussion by gotenxds** [samchon/typia#1253] [Q&A] [answered]
Hello, is would like to create a type that is required based on the value of a different type.

in JSON schema you could do this like so: 

```js
         x: {
          type: 'object',
          properties: {
            value: {type: 'boolean'},
            range: {type: 'integer', multipleOf: 5, minimum: 0, maximum: 60}
          },
          if: {
            properties: {
              value: {const: true}
            }
          },
          then: {required: ['range']}
}
```

In typia, to the best of my ability I was able to create something similar using unions
```ts
x: { value: false; range?: number } | { value: true; range: number }
```

But this is not the same, the generated json-schema will be a `oneOf` and the const value of 'true' is generated like so `{type: boolean; enum: [true]}` which is a really odd way of doing it.

is this at all possible with typia?

**Comment by samchon** [comment, marked as answer]
Not possible, because TypeScript does not have the if condition type spec.

Instead, the union (`oneOf`) type will cover what you want.

**Reply by gotenxds** [reply]
Thanks @samchon, the `oneof` will not cover what I want, they are not functionally the same, the json schema resulted is not the same and is interpreted differently in different tools.

Guess for now I'l stick with TypeBox which do support it.

**Comment by samchon** [comment]
Closing as answered by the maintainer.

TypeScript does not have a direct `if` / `then` type construct, so typia cannot emit JSON Schema `if` / `then` from the source type. The TypeScript-native representation is the union you already wrote:

```ts
type X = { value: false; range?: number } | { value: true; range: number };
```

That naturally becomes a union schema (`oneOf`). Representing a boolean literal as `enum: [true]` is a valid schema representation for the literal branch.
